source/examples/bluetooth-eddystone-url/main.cpp

```diff
@@ -0,0 +1,51 @@
+#include "MicroBit.h"
+
+MicroBit uBit;
+
+char URL[] = "https://goo.gl/TlUTF7";
+const int8_t CALIBRATED_POWERS[] = {-49, -37, -33, -28, -25, -20, -15, -10};
+
+uint8_t advertising = 0;
+uint8_t tx_power_level = 6;
+
+void startAdvertising() {
+    uBit.bleManager.advertiseEddystoneUrl(URL, CALIBRATED_POWERS[tx_power_level-1], false);
+    uBit.bleManager.setTransmitPower(tx_power_level);
+    uBit.display.scroll("ADV");
+    advertising = 1;
+}
+
+void stopAdvertising() {
+    uBit.bleManager.stopAdvertising();
+    uBit.display.scroll("OFF");
+    advertising = 0;
+}
+
+void onButtonA(MicroBitEvent)
+{
+    if (advertising == 1) {
+        return;
+    }
+    startAdvertising();
+}
+
+void onButtonB(MicroBitEvent)
+{
+    if (advertising == 0) {
+        return;
+    }
+    stopAdvertising();
+}
+
+int main()
+{
+    // Initialise the micro:bit runtime.
+    uBit.init();
+
+    uBit.messageBus.listen(MICROBIT_ID_BUTTON_A, MICROBIT_EVT_ANY, onButtonA);
+    uBit.messageBus.listen(MICROBIT_ID_BUTTON_B, MICROBIT_EVT_ANY, onButtonB);

```

Comment:
Hi @bluetooth-mdw Can we change the MICROBIT_EVT_ANY in these two lines here to something more specific (e.g. MICROBIT_EVT_BUTTON_CLICK)? Your event handler will fire far more than you intend otherwise. While not a problem for this code, being a sample, people will likely copy it and be a bit surprised by the result. :-)